internal/filter/match.go

```diff
@@ -137,11 +137,10 @@ func (s ClauseSet) match(c Clause, objValue any) (bool, error) {
 	case reflect.Bool:
 		valueBool, err = s.ParseBool(c)
 	case reflect.Slice:
-		if reflect.TypeOf(objValue).Elem().Kind() == reflect.String {
-			valueSlice, err = s.ParseStringSlice(c)
-		} else {
+		if reflect.TypeOf(objValue).Elem().Kind() != reflect.String {
 			return false, fmt.Errorf("Invalid slice type %q for field %q", reflect.TypeOf(objValue).Elem().Kind(), c.Field)
 		}
+		valueSlice, err = s.ParseStringSlice(c)
```

Comment:
There should be a line break before this line

Reply:
Done